source/actions.js

```diff
@@ -162,35 +167,48 @@ export const watchEvent = (firebase, dispatch, event, path, dest, onlyLastEvent
   }
 
   const runQuery = (q, e, p) => {
+      dispatch({
+        type: START,
+        timestamp: moment(),
+        requesting : true,
+        requested : false,
+        path
+      })
+  
     q.on(e, snapshot => {
       let data = (e === 'child_removed') ? undefined : snapshot.val()
-      const resultPath = dest || (e === 'value') ? p : p + '/' + snapshot.key()
+      const resultPath = (dest) ? dest : (e === 'value') ? p : p + '/' + snapshot.key
+      const rootPath = (dest) ? dest :  path
```

Comment:
i changed that because on the original code watch function was send "true" to dest and then you will get resultPath=true and thats wrong, after i removed the true from dest we can remove this to.